lib/Parser/Parser.cpp

```diff
@@ -1122,6 +1131,74 @@ bool Parser::parseLine(std::string &ErrStr) {
                   }
                   if (!consumeToken(ErrStr))
                     return false;
+                } else if (CurTok.str() == "range") {
+                  // look for =
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::Eq) {
+                    ErrStr = makeErrStr(TP, "expected '=' for range as 'range='");
+                    return false;
+                  }
+
+                  // look for [ (inclusive paren)
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::OpenBracket) {
+                    ErrStr = makeErrStr(TP, "expected '[' to specify lower bound of range");
+                    return false;
+                  }
+
+                  // look for untypedconst
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::UntypedInt && CurTok.K == Token::Int) {
+                    ErrStr = makeErrStr(TP, "expected lower bound of range without width");
+                    return false;
+                  }
+                  if (CurTok.K != Token::UntypedInt) {
+                    ErrStr = makeErrStr(TP, "expected lower bound of range");
+                    return false;
+                  }
+                  Lower = CurTok.Val;
+
+                  if (Lower.getBitWidth() != InstWidth) {
+                    Lower = Lower.sextOrTrunc(InstWidth);
+                  }
+
+                  // look for comma
```

Comment:
remove all of these comments, which are about things that are pretty obvious from looking at the code